test/caPutJsonLogTest.cpp

```diff
@@ -812,6 +851,39 @@ void runTests(void *arg) {
            &charArray1[0], strlen(charArray1), &charArray2[0], strlen(charArray1), "DBF_CHAR array (lso) test");
 
 
+    // Test metadata
+    std::map<std::string, std::string> metadata_test_single = {
+        {"ABCDEFGHIJKLMNOPQRS", "1234567890_1234567890"}
+    };
+    std::map<std::string, std::string> metadata_test_escape_chars = {
+        {"ABC\n##&¤67&nbsp; space&#160&&#160;\'\\n#", "CBC\n##&¤67&nbsp; space&#160&&#160;\'\\n#"}
+    };
+    std::map<std::string, std::string> metadata_test_multiple = {
+        {"One", "1"}, {"Two", "2"}, {"Three", "3"}, {"Four", "4"}, {"Five", "5"},
+        {"Six", "6"}, {"Seven", "7"}, {"Eight", "8"}, {"Nine", "9"}, {"Ten", "10"},
+    };
+
+    std::map<std::string, std::string> metadata_test_reinsert_new_val = {
+        {"Five", "5"}, {"Six", "6"},
+    };
+
+    testMetadataHelper(metadata_test_single);
+    testMetadataHelper(metadata_test_escape_chars);
+    testMetadataHelper(metadata_test_multiple);
+
+    //Reinsert metadata test
+    logger->addMetadata("Five", "Five");
+    logger->addMetadata("Six", "Six");
+    testMetadataHelper(metadata_test_reinsert_new_val);
```

Comment:
This does not reinsert because at end of addMetadata you remove all metadata.

Reply:
I meant testMetadataHelper, sorry.

> testMetadataHelper does at the end, but before that we have test there that check if what's in the logger matches what was in the test map used to insert the new values.

I see, but if it removes all metadata at the end how lines 875 and 876 are doing a reinsert?

Reply:
They do the first insert. The re-insert is done via testMetadataHelper, using the new values provided by metadata_test_reinsert_new_val

Reply:
Ohh I miss that. Ok.